Source/JavaScriptCore/inspector/agents/InspectorDebuggerAgent.cpp

```diff
@@ -1460,4 +1576,14 @@ void InspectorDebuggerAgent::clearAsyncStackTraceData()
     didClearAsyncStackTraceData();
 }
 
+bool InspectorDebuggerAgent::SymbolicBreakpoint::matches(const String& symbol) const
+{
+    if (symbol.isEmpty())
+        return false;
+
+    auto searchStringType = isRegex ? ContentSearchUtilities::SearchStringType::Regex : ContentSearchUtilities::SearchStringType::ExactString;
+    auto regex = ContentSearchUtilities::createRegularExpressionForSearchString(this->symbol, caseSensitive, searchStringType);
+    return regex.match(symbol) != -1;
```

Comment:
nit: Should we just cache the RegularExpression on the SymbolicBreakpoint struct? We might be calling this a lot